### PR TITLE
[onert] Introduce CompilerArtifact

### PR DESCRIPTION
**Pull request by hseok-oh**
This commit introduceCompilerArtifact class.
Compiler returns CompilerArtifact which includes ExecutorMap and TracingCtx.

Signed-off-by: Hyeongseok Oh <hseok82.oh@samsung.com>

---

Related issue: #9281
Draft: #9280

### CODE REVIEW

**Comment by hseok-oh** [comment]
@nnfw-bot test cross-aarch64-runtime-test-release
@nnfw-bot test cross-runtime-test-debug 
@nnfw-bot test cross-runtime-test-release

File: runtime/onert/core/include/compiler/Compiler.h [hunk 1]
```diff
@@ -101,18 +113,18 @@ class Compiler
   /**
    * @brief   Do compilation with the options
    *
-   * @return std::shared_ptr<exec::ExecutorMap> Executors as a result of compilation
+   * @return std::shared_ptr<CompilationResult> Executors as a result of compilation
```
Comment (glistening) [inline runtime/onert/core/include/compiler/Compiler.h:116]:
```suggestion
   * @return std::shared_ptr<CompilerArtifact> Executors as a result of compilation
```

File: runtime/onert/core/include/compiler/Compiler.h [hunk 2]
```diff
@@ -101,18 +113,18 @@ class Compiler
...
    */
-  std::shared_ptr<exec::ExecutorMap> compile(void);
+  std::shared_ptr<CompilerArtifact> compile(void);
 
   /**
    * @brief   Do compilation with the options
    *
-   * @return std::vector<std::shared_ptr<exec::ExecutorMap>> Executors as a result of compilation
+   * @return std::vector<std::shared_ptr<CompilationResult>> Executors as a result of compilation
```
Comment (glistening) [inline runtime/onert/core/include/compiler/Compiler.h:123]:
```suggestion
   * @return std::vector<std::shared_ptr<CompilerArtifact>> Executors as a result of compilation
```

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 3]
```diff
@@ -361,25 +361,28 @@ std::shared_ptr<exec::ExecutorMap> Compiler::compile(void)
...
-    return executors;
+    return std::make_shared<CompilerArtifact>(executors, nullptr);
   }
 
   // Mode check
   if (_options.he_profiling_mode)
     checkProfilerConditions();
 
+  // Tracing context
+  auto tracing_ctx = std::make_unique<util::TracingCtx>(_subgraphs.get());
```
Comment (glistening) [inline runtime/onert/core/src/compiler/Compiler.cc:384]: @hseok-oh (beyond this PR) What about creating `tracing_ctx` only tracing is enabled (i.e. `TRACEFILE_PATH` != nullptr)?

**Review by chunseoklee — DISMISSED** [review]

File: runtime/onert/core/src/compiler/Compiler.cc [hunk 4]
```diff
@@ -559,12 +563,15 @@ std::vector<std::shared_ptr<exec::ExecutorMap>> Compiler::compile(const char *pa
   //       execution between interpreter and compiled executor (including control flow)
   if (_options.disable_compile)
   {
+    std::vector<std::shared_ptr<CompilerArtifact>> results;
```
Comment (glistening) [inline runtime/onert/core/src/compiler/Compiler.cc:566]: @chunseoklee PTAL. As I understand, you are going to introduce some class for collection of `ExecutorMap`.

File: runtime/onert/frontend/nnapi/wrapper/ANeuralNetworksCompilation.h [hunk 5]
```diff
@@ -36,7 +36,13 @@ struct ANeuralNetworksCompilation
...
   void publish(std::shared_ptr<onert::exec::ExecutorMap> &executors) noexcept
   {
-    executors = _executors;
+    if (_artifact == nullptr)
+    {
+      executors = nullptr;
+      return;
+    }
+
+    executors = _artifact->_executors;
```
Comment (glistening) [inline runtime/onert/frontend/nnapi/wrapper/ANeuralNetworksCompilation.h:45]:
```suggestion
    executors = _artifact ? _artifact->_executors : nullptr;
```

File: runtime/onert/core/src/compiler/Compiler.cc (hunk 3 above)
Comment (hseok-oh) [reply runtime/onert/core/src/compiler/Compiler.cc:384]: Good. I'll update later.

File: runtime/onert/core/src/compiler/Compiler.cc (hunk 4 above)
Comment (chunseoklee) [reply runtime/onert/core/src/compiler/Compiler.cc:566]: Not decided yet. I will take care of it again after this PR is merged. At this time, this looks fine so far.

**Review by glistening — APPROVED** [review]
LGTM

**Review by chunseoklee — APPROVED** [review]

**Pull request merged**